src/components/lyrics.tsx

```diff
@@ -1,74 +1,88 @@
-          const response = await fetch(`/api/lyrics/${encodeURIComponent(lyricsLink)}`);
-          if (!response.ok) {
-            throw new Error(`Response status: ${response.status}`);
-          }
-    
-          const data = await response.json();
-          if(data.lyrics){
-          setLyrics(data.lyrics);
-          await translateLyrics(data.lyrics);
-        } 
-        } catch (error) {
-          console.error("Error fetching lyrics:", error);
-        }
-      };
-    
+const translateLyrics = async (lyrics: string) => {
+  try {
+    const response = await fetch("/api/translate", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        text: lyrics,
+        targetLang: "pl",
+      }),
+    });
 
-  const translateLyrics = async (lyrics: string) => {
-    try {
-      const response = await fetch("/api/translate", {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify({
-          text: lyrics,
-          targetLang: "pl",
-        }),
-      });
-  
-      const data = await response.json();
-  
-      if (data.translatedText) {
-        setTranslatedLyrics(data.translatedText);
-      } 
-    } catch (error) {
-      console.error("Error translating lyrics:", error);
-    }
-  };
+    const data = await response.json();
+    return data.translatedText;
+  } catch (error) {
+    console.error("Error translating lyrics:", error);
+  }
+};
+
+export default function Lyrics({ lyricsLink }: { lyricsLink: string }) {
+  const [translatedLyrics, setTranslatedLyrics] = useState<string | null>(null);
+  const [lyrics, setLyrics] = useState<string | null>(null);
+  const [isLoadingLyrics, setIsLoadingLyrics] = useState(true);
+  const [isLoadingTranslatedLyrics, setIsLoadingTranslatedLyrics] = useState(true);
+  const [requestMade, setRequestMade] = useState(false);
 
   useEffect(() => {
+
+    const makeRequest = async () => {
+      const lyrics = await fetchLyrics(lyricsLink);
+      setLyrics(lyrics);
+      setIsLoadingLyrics(false);
+
+      if (lyrics) {
+        const translatedLyrics = await translateLyrics(lyrics);
+        setTranslatedLyrics(translatedLyrics);
+        setIsLoadingTranslatedLyrics(false);
+      }
+    };
+
     if (lyricsLink) {
-      fetchLyrics();
+      makeRequest();
+      setRequestMade(true);
     }
-  }, [lyricsLink]);
+  }, [lyricsLink, requestMade]);
 
   return (
-        <div>
-          <h1>Lyrics Viewer</h1>
-          {lyrics ? (
-            <div>
-              <h2>Original Lyrics</h2>
-              <p>{lyrics}</p>
-            </div>
-          ) : (
-            <p>Loading lyrics...</p>
-          )}
-    
-          {translatedLyrics && (
-            <div>
-              <h2>Translated Lyrics (Polish):</h2>
-              <p>{translatedLyrics}</p>
-            </div>
-          )}
+    <div>
+      <h2>Lyrics Viewer</h2>

```

Comment:
make design more pretty, text goes outside the container 
<img width="1123" alt="image" src="https://github.com/user-attachments/assets/6a574418-3bb3-43e3-be91-ddcef6ffbf26">
it shouldnt look like this